File: internal/domain/document/model/entity/document.go
package entity

import (
	"time"

	"github.com/swiftbit/know-agent/internal/domain/document/model/vo"
)

type Document struct {
	ID                   int64     `gorm:"column:id"`                    // 主键ID
	DocumentName         string    `gorm:"column:document_name"`         // 文档名称
	OriginalFileName     string    `gorm:"column:original_file_name"`    // 原始文件名
	FileType             int       `gorm:"column:file_type"`             // 文件类型
	MimeType             string    `gorm:"column:mime_type"`             // 媒体类型
	FileSize             int64     `gorm:"column:file_size"`             // 文件大小
	StorageType          int       `gorm:"column:storage_type"`          // 存储类型
	BucketName           string    `gorm:"column:bucket_name"`           // 存储桶名称
	ObjectName           string    `gorm:"column:object_name"`           // 对象名称
	ObjectUrl            string    `gorm:"column:object_url"`            // 对象访问地址
	ParseStatus          int       `gorm:"column:parse_status"`          // 解析状态
	StrategyStatus       int       `gorm:"column:strategy_status"`       // 策略状态
	IndexStatus          int       `gorm:"column:index_status"`          // 索引状态
	CharCount            int       `gorm:"column:char_count"`            // 字符数
	TokenCount           int       `gorm:"column:token_count"`           // Token数量
	StructureLevel       int       `gorm:"column:structure_level"`       // 结构化等级
	ContentQualityLevel  int       `gorm:"column:content_quality_level"` // 内容质量等级
	ParseTextPath        string    `gorm:"column:parse_text_path"`       // 解析文本路径
	ParseErrorMsg        string    `gorm:"column:parse_error_msg"`       // 解析错误信息
	KnowledgeScopeCode   string    `gorm:"column:knowledge_scope_code"`  // 知识范围编码
	KnowledgeScopeName   string    `gorm:"column:knowledge_scope_name"`  // 知识范围名称
	BusinessCategory     string    `gorm:"column:business_category"`     // 业务分类
	DocumentTags         string    `gorm:"column:document_tags"`         // 文档标签
	CurrentPlanId        int64     `gorm:"column:current_plan_id"`       // 当前方案ID
	LastParseTaskId      int64     `gorm:"column:last_parse_task_id"`    // 上一次解析任务ID
	StructureNodeCount   int       `gorm:"column:structure_node_count"`  // 结构化节点数
	LastIndexTaskId      int64     `gorm:"column:last_index_task_id"`    // 上一次索引任务ID
	CreateTime           time.Time `gorm:"column:create_time"`           // 创建时间
	UpdateTime           time.Time `gorm:"column:update_time"`           // 更新时间
	OperatorId           int64     `gorm:"-"`                            // 操作人ID
	FileTypeName         string    `gorm:"-"`                            // 文件类型名称
	ParseStatusName      string    `gorm:"-"`                            // 解析状态名称
	StrategyStatusName   string    `gorm:"-"`                            // 策略状态名称
	IndexStatusName      string    `gorm:"-"`                            // 索引状态名称
	LatestTaskId         int64     `gorm:"-"`                            // 最新任务ID
	LatestTaskType       int       `gorm:"-"`                            // 最新任务类型
	LatestTaskTypeName   string    `gorm:"-"`                            // 最新任务类型名称
	LatestTaskStatus     int       `gorm:"-"`                            // 最新任务状态
	LatestTaskStatusName string    `gorm:"-"`                            // 最新任务状态名称
	PlanReady            bool      `gorm:"-"`                            // 方案是否就绪
}

func (d *Document) FillEnumNames() {
	d.FileTypeName = vo.FileTypeName(d.FileType)
	d.ParseStatusName = vo.ParseStatusName(d.ParseStatus)
	d.StrategyStatusName = vo.StrategyStatusName(d.StrategyStatus)
	d.IndexStatusName = vo.IndexStatusName(d.IndexStatus)
}

func (d *Document) FillLatestTaskInfo(task *DocumentTask) {
	d.LatestTaskId = task.ID
	d.LatestTaskType = task.TaskType
	d.LatestTaskTypeName = vo.TaskTypeName(task.TaskType)
	d.LatestTaskStatus = task.TaskStatus
	d.LatestTaskStatusName = vo.TaskStatusName(task.TaskStatus)
}

File: src/types/document.ts
// 文档类型
/** 上传文档请求 */
export interface UploadDocumentReq {
  documentName?: string;
  operatorId?: string;
  knowledgeScopeCode?: string;
  knowledgeScopeName?: string;
  businessCategory?: string;
  documentTags?: string;
  file: File | null;
}

/** 分页查询文档列表请求 */
export interface QueryDocumentPageReq {
  pageNo?: number;
  pageSize?: number;
  keyword?: string;
}

/** 查询文档详情请求 */
export interface QueryDocumentDetailReq {
  documentId: string;
}

/** 删除文档请求 */
export interface DeleteDocumentReq {
  documentId: string;
}

/** 查询文档策略推荐结果请求 */
export interface QueryStrategyPlanReq {
  documentId: string;
}

/** 策略步骤子项（公共内嵌实体） */
export interface StrategyStepItem {
  stepNo: number;
  strategyType: 1 | 2 | 3;
}

/** 确认文档策略方案请求 */
export interface ConfirmStrategyReq {
  documentId: string;
  basePlanId: string;
  operatorId?: string;
  adjustNote?: string;
  parentSteps: StrategyStepItem[];
  childSteps: StrategyStepItem[];
}

/** 执行文档索引构建请求 */
export interface BuildIndexReq {
  documentId: string;
  planId: string;
  operatorId?: string;
}

/** 查询文档chunk列表请求 */
export interface QueryDocumentChunksReq {
  documentId: string;
  taskId?: string;
  pageNo?: number;
  pageSize?: number;
}

/** 查询单个文档chunk详情请求 */
export interface QueryDocumentChunkDetailReq {
  documentId: string;
  chunkId: string;
  taskId?: string;
}

/** 查询任务执行日志请求 */
export interface QueryTaskLogsReq {
  taskId: string;
  pageNo?: number;
  pageSize?: number;
}

/** 查询文档画像详情请求 */
export interface DocumentProfileDetailReq {
  documentId: string;
}

/** 重新生成文档画像请求 */
export interface DocumentProfileRegenerateReq {
  documentId: string;
  operatorId?: string;
}

/** 批量重新生成文档画像请求 */
export interface DocumentProfileBatchRegenerateReq {
  documentIds: string[];
  operatorId?: string;
}

// ====================== 内嵌基础响应子类型 ======================
/** 策略步骤 */
export interface DocumentStrategyStep {
  stepNo: number;
  pipelineType: string;
  pipelineTypeName: string;
  strategyType: number;
  strategyTypeName: string;
  strategyRole: number;
  strategyRoleName: string;
  sourceType: number;
  sourceTypeName: string;
  executeStatus: number;
  executeStatusName: string;
  recommendReason: string;
}

/** 策略流水线 */
export interface DocumentStrategyPipeline {
  pipelineType: string;
  pipelineTypeName: string;
  strategySnapshot: string;
  steps: DocumentStrategyStep[];
}

/** 策略方案 */
export interface DocumentStrategyPlan {
  planId: string;
  planVersion: number;
  planSource: number;
  planSourceName: string;
  planStatus: number;
  planStatusName: string;
  strategySnapshot: string;
  recommendReason: string;
  parentPipeline: DocumentStrategyPipeline | null;
  childPipeline: DocumentStrategyPipeline | null;
}

/** 文档父块 */
export interface DocumentParentBlockItem {
  parentBlockId: string;
  parentBlockNo: number;
  sectionPath: string;
  sourceType: number;
  sourceTypeName: string;
  charCount: number;
  tokenCount: number;
  childCount: number;
  startChunkNo: number;
  endChunkNo: number;
  parentText: string;
}

/** 文档分片 */
export interface DocumentChunkItem {
  chunkId: string;
  parentBlockId: string;
  parentBlockNo: number;
  parentChildCount: number;
  parentStartChunkNo: number;
  parentEndChunkNo: number;
  chunkNo: number;
  sectionPath: string;
  sourceType: number;
  sourceTypeName: string;
  charCount: number;
  tokenCount: number;
  vectorStatus: number;
  vectorStatusName: string;
  chunkText: string;
}

/** 任务日志 */
export interface TaskLog {
  id: string;
  stageType: number;
  stageTypeName: string;
  eventType: number;
  eventTypeName: string;
  logLevel: number;
  logLevelName: string;
  content: string;
  detailJson: string;
  createTime: string;
}

// ====================== 顶层响应类型 ======================
/** 上传文档响应 */
export interface UploadDocumentResp {
  documentId: string;
  taskId: string;
  documentName: string;
  parseStatus: number;
  strategyStatus: number;
  indexStatus: number;
}

/** 文档详情 */
export interface DocumentDetailResp {
  documentId: string;
  documentName: string;
  originalFileName: string;
  fileType: number;
  fileTypeName: string;
  fileSize: number;
  charCount: number;
  tokenCount: number;
  parseStatus: number;
  parseStatusName: string;
  strategyStatus: number;
  strategyStatusName: string;
  indexStatus: number;
  indexStatusName: string;
  parseErrorMsg: string;
  knowledgeScopeCode: string;
  knowledgeScopeName: string;
  businessCategory: string;
  documentTags: string;
  currentPlanId: string;
  lastIndexTaskId: string;
  latestTaskId: string;
  latestTaskType: number;
  latestTaskTypeName: string;
  latestTaskStatus: number;
  latestTaskStatusName: string;
  createTime: string;
  updateTime: string;
}

/** 分页查询文档列表响应 */
export interface QueryDocumentPageResp {
  pageNo: number;
  pageSize: number;
  total: number;
  records: DocumentDetailResp[];
}

/** 文档知识库选项响应 */
export interface KnowledgeDocumentOptionResp {
  documentId: string;
  documentName: string;
  knowledgeScopeName: string;
  businessCategory: string;
  documentTags: string[];
}

/** 删除文档响应 */
export interface DeleteDocumentResp {
  documentId: string;
  documentName: string;
}

/** 查询文档策略方案响应 */
export interface QueryStrategyPlanResp {
  documentId: string;
  documentName: string;
  parseStatus: number;
  parseStatusName: string;
  strategyStatus: number;
  strategyStatusName: string;
  indexStatus: number;
  indexStatusName: string;
  parseErrorMsg: string;
  planReady: boolean;
  plan: DocumentStrategyPlan | null;
}

/** 确认文档策略方案响应 */
export interface ConfirmStrategyResp {
  planId: string;
  documentId: string;
  planVersion: number;
  strategyStatus: number;
  strategyStatusName: string;
  normalized: boolean;
  parentPipeline: DocumentStrategyPipeline | null;
  childPipeline: DocumentStrategyPipeline | null;
}

/** 执行文档索引构建响应 */
export interface BuildIndexResp {
  taskId: string;
  documentId: string;
  taskType: number;
  taskTypeName: string;
  taskStatus: number;
  taskStatusName: string;
  indexStatus: number;
  indexStatusName: string;
}

/** 查询文档chunk列表响应 */
export interface QueryDocumentChunksResp {
  documentId: string;
  taskId: string;
  planId: string;
  pageNo: number;
  pageSize: number;
  total: number;
  records: DocumentChunkItem[];
}

/** 查询单个文档chunk详情响应 */
export interface QueryDocumentChunkDetailResp {
  documentId: string;
  taskId: string;
  planId: string;
  chunk: DocumentChunkItem | null;
  parentBlock: DocumentParentBlockItem | null;
  siblingChunks: DocumentChunkItem[];
}

/** 查询任务执行日志响应 */
export interface QueryTaskLogsResp {
  taskId: string;
  documentId: string;
  taskType: number;
  taskTypeName: string;
  taskStatus: number;
  taskStatusName: string;
  currentStage: number;
  currentStageName: string;
  startTime: string;
  finishTime: string;
  costMillis: number;
  errorCode: string;
  errorMsg: string;
  total: number;
  logs: TaskLog[];
}

/** 文档画像详情响应 */
export interface DocumentProfileResp {
  documentId: string;
  documentSummary: string;
  documentType: string;
  coreTopics: string;
  exampleQuestions: string;
  graphFriendly: number;
  supportsGraphOutline: number;
  supportsItemLookup: number;
  supportsGraphAssist: number;
  profileSource: string;
  profileStatus: number;
  errorMsg: string;
}

